app/components/issues/badge.html.erb

```diff
@@ -0,0 +1,3 @@
+<%= render(Primer::State.new(color: STATES[state][:color], title: "Status: #{state.to_s.titleize}")) do %>
```

Comment:
When the template was defined in the `issues/badge.rb` file it was easier to tell that `STATES` was referencing the lexically-scoped `Issues::Badge::STATES`.

Moving it to a sidecar file makes that less evident, which makes me wonder if you'd recommend not relying on the lexical scope of the ruby class when using a separate template file (e.g. would it be recommended to replace `STATES` with the fully-qualified `Issues::Badge::STATES` here)?

Reply:
@rmacklin that's a good point, and I think a fully scoped lookup would be more clear. 

This implementation still allows for templates to be defined inside the component file, as `Box` and `StringInterp` in this repo do. 

The main reason we've switched to sidecars as our default is that a lot of editors didn't play nicely with the HEREDOCs.

I'm guessing which one is used will be a matter of taste.